realm/realm-library/src/main/cpp/objectserver_shared.hpp

```diff
@@ -81,6 +82,9 @@ class JniSession {
                             local_env, obj, "notifySessionError", "(ILjava/lang/String;)V");
                     local_env->CallVoidMethod(
                             obj, notify_error_handler, error_code.value(), local_env->NewStringUTF(message.c_str()));
+                    if (local_env->ExceptionCheck() == JNI_TRUE) {
+                        throw realm::jni_util::JniPendingException("Exception in error handler");
```

Comment:
Good point. But I imagine that if an error handler throws an exception, the app will terminate. The reason for this code change is that we wish the app developer to see the exception and not some random JNI error.